src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -266,6 +266,36 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanC
                 }
             }
 
+            /// <summary>
+            /// Force computation of up to date tags.
+            /// This is called by the editor for scenarios such as print or rich text copy with classifications enabled.
+            /// Note that the editor will block the UI until results are returned.
+            /// </summary>
+            public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSpanCollection spans, CancellationToken cancellationToken)
+            {
+                if (spans.IsEmpty() || _workspace == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ITagSpan<IClassificationTag>>();
+                }
+
+                var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                var classificationService = document?.Project.LanguageServices.GetService<IClassificationService>();
+                if (classificationService == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ITagSpan<IClassificationTag>>();
+                }
+
+                var classifiedSpans = ClassificationUtilities.GetOrCreateClassifiedSpanList();
+                foreach (var span in spans)
+                {
+                    classificationService.AddSyntacticClassificationsAsync(
+                        document, span.Span.ToTextSpan(), classifiedSpans, cancellationToken).Wait(cancellationToken);
+                }
```

Comment:
Any reason this shouldn't just call AddClassifiedSpansForCurrentTree? That way the logic is shared, and we'll reuse the cache if it is already up to date?

Reply:
Also I can understand if the answer is "no", because if the cache is still on an older snapshot we might confuse the cache in some way.

Reply:
I think I can do this, just need to check that the last snapshot matches before attempting to do AddClassifiedSpansForCurrentTree